### PR TITLE
Feature/auth model

### PR DESCRIPTION
**Pull request by shamoons**

### CODE REVIEW

File: models/role.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module.exports = (sequelize, DataTypes) => {
+  const Role = sequelize.define(
```
Comment (gopalakrishnanu) [inline models/role.js]: Role-privileges association is misssing

Comment (gopalakrishnanu) [reply models/role.js]: UserRole model is missing.

File: routes/index.js [hunk 2]
```diff
@@ -1,23 +1,9 @@
...
-      '/api/users',
-      '/api/users/:id'
-    ],
-    associations: true,
-    pagination: false
-  })
-
-  userResource.list.fetch.before((req, res, context) => {
-    passport.authenticate('jwt', { session: false })
+  app.get("/users/:id", passport.authenticate('jwt', { session: false }), (req, res) => {
```
Comment (gopalakrishnanu) [inline routes/index.js]: Can we move all auth related to its own auth.js file? Each model routes will be on its own route files.

File: models/user.js [hunk 3]
```diff
@@ -2,21 +2,45 @@ module.exports = (sequelize, DataTypes) => {
...
+        type: DataTypes.DATE
       },
-      first_name: {
-        type: DataTypes.TEXT
+      active: {
+        type: DataTypes.BOOLEAN
       },
-      role: {
-        type: DataTypes.TEXT
+      lastName: {
```
Comment (gjanakiram) [inline models/user.js]: added lastName twice ?

**Comment by gjanakiram** [comment]
@shamoons So we are not connecting to postgres to get user information right ?

**Comment by shamoons** [comment]
@gjanakiram we are.  As per the comment at https://github.com/semanticbits/qio-api/pull/4#discussion_r155652725 we are now pulling data from the DB. But the DB is seeded with some default user information

**Comment by gjanakiram** [comment]
Excellent. Thanks @shamoons

**Pull request merged**